### DISCUSSION TITLE
Pre-packaged version(s) of wifibox

### DISCUSSION THREAD

**Discussion by MegaManSec** [pgj/freebsd-wifibox#116] [Ideas]
At the moment, the installation of wifibox requires an internet connection to download the various github release packages.

It would be extremely helpful if a single package could be used to install wifibox onto a system (such as via a USB).

Alternatively, an AIO guide for building and installing from the Github repositories would be helpful.

**Comment by pgj** [comment]
Although it is not strictly required, the Wifibox script, its configuration files, and the VM image could be installed through using the `net/wifibox-core` and `net/wifibox-alpine` ports and the corresponding binary packages.  There is the `net/wifibox` port as well, but that is just a top-level meta-port or -package and as such it is not mandatory to get, its purpose is only to make it easier to install the other two ports/packages.

The way how these ports should be installed follows the established conventions, but there is a README file in the [pgj/freebsd-wifibox-port](https://github.com/pgj/freebsd-wifibox-port) repository that captures everything that one should know about them.  Though I tried to make everything automatic that is possible, e.g. checking for the presence of the `linux64` kernel module on building the `net/wifibox-alpine` port.  The whole idea behind using ports is to save the users from dealing with the laborious bits of compiling and installing the parts.  But, for modularity and the flow development, the source files are kept in different `git` repositories and not in the port files.

Other than that, one could always leverage the `pkg` tool to get prepared for an off-line install.  For example, use `pkg fetch --dependencies wifibox` to download all of required packages upfront which could be then copied over to an USB disk.  These packages could be then fed into `pkg add` during the offline install.

However, let me note that Wifibox is included in the FreeBSD Ports Collection and receives updates every second quarter (Q2 and Q4) so if one actually downloads a FreeBSD installer with binary packages included (DVD format) it might be possible that the image has Wifibox as well.  (If not, we could get in touch with the Ports Management Team to fix this.)

**Reply by MegaManSec** [reply]
>However, let me note that Wifibox is included in the FreeBSD Ports Collection and receives updates every second quarter (Q2 and Q4) so if one actually downloads a FreeBSD installer with binary packages included (DVD format) it might be possible that the image has Wifibox as well. (If not, we could get in touch with the Ports Management Team to fix this.)

It seems it is not:

```
$ ls /mnt/packages/FreeBSD\:13\:amd64/All/wifi*
ls: cannot access '/mnt/packages/FreeBSD:13:amd64/All/wifi*': No such file or directory
$ ls /mnt/packages/FreeBSD\:14\:amd64/All/wifi*
ls: cannot access '/mnt/packages/FreeBSD:14:amd64/All/wifi*': No such file or directory
```
